cpp/include/cudf/io/functions.hpp

```diff
@@ -428,11 +433,149 @@ struct read_parquet_args {
   /// Cast timestamp columns to a specific type
   data_type timestamp_type{type_id::EMPTY};
 
+  explicit read_parquet_args(source_info const& src) : source(src) {}
+
+  friend read_parquet_args_builder;
+
+ public:
   explicit read_parquet_args() = default;
 
-  explicit read_parquet_args(source_info const& src) : source(src) {}
+  /**
+   * @brief create read_parquet_args_builder which will build read_parquet_args
+   *
+   * @param src source information used to read parquet file
+   * @returns read_parquet_args_builder
+   */
+  static read_parquet_args_builder build(source_info const& src);
+
+  /**
+   * @brief Returns source info
+   */
+  source_info get_source_info() const { return source; }
+
+  /**
+   * @brief Returns column names
+   */
+  std::vector<std::string> column_names() const { return columns; }
+
+  /**
+   * @brief Returns row groups
+   */
+  std::vector<std::vector<size_type>> get_row_groups() const { return row_groups; }
+
+  /**
+   * @brief Returns number of rows to skip
+   */
+  size_type rows_to_skip() const { return skip_rows; }
+
+  /**
+   * @brief Returns row groups
+   */
+  size_type get_num_rows() const { return num_rows; }
+
+  /**
+   * @brief Returns strings_to_categorical
+   */
+  bool is_strings_to_categorical() const { return strings_to_categorical; }
```

Comment:
`strings_are_categorical` would read more naturally. Or maybe `are_strings_categorical`? @codereport what's the precedent for using `are` in a predicate?

Reply:
The typical ones i have seen are `is_` and `has_` but `are_` is the plural of `is_` so I think that is fine. This is where it would be nice to have the LISP convention of using `?`s at the end of predicates, so we could have `strings_categorical?`. 

As an additional comment, I don't understand what any of the function names mean, is there a comment explaining what `strings_to_categorical` means somewhere?

Reply:
https://github.com/rapidsai/cudf/blob/6140a4bda70676f3ecdc865a03a247ae27be57f7/cpp/include/cudf/io/functions.hpp#L429

Reply:
`get(STRINGS_TO_CATEGORICAL)`